### DISCUSSION TITLE
trying to standardize image orientation using Orientationd transform

### DISCUSSION THREAD

**Discussion by jakubMitura14** [Project-MONAI/MONAI#3906] [Q&A] [answered]
Hello I am preprocessing CT-ORG [1] dataset - and I want to make orientation of all images uniform.  As  far as I understand this should be achievable using Orientationd Transform - However as I will show above for my case once the bed is on the left and other time on the right side of image. For this example I am loading just patient number 20 and 21.

![image](https://user-images.githubusercontent.com/53857487/157379774-f39be42f-c68a-4008-915f-54914f8b5414.png)

defined transforms 

```
val_transforms = Compose(
    [
        LoadImaged(keys=["image", "label"]),
        EnsureChannelFirstd(keys=["image", "label"]),
        Orientationd(keys=["image", "label"], axcodes="RAS"),
        Spacingd(keys=["image", "label"], pixdim=(
            1.0, 1.0, 1.0), mode=("bilinear", "nearest")),
        CropForegroundd(keys=["image", "label"], source_key="image"),
        EnsureTyped(keys=["image", "label"]),
    ]
)

```

visualization code 
```
check_ds = Dataset(data=data_dicts, transform=val_transforms)
check_loader = DataLoader(check_ds, batch_size=1)
for dat in check_loader:
    image= dat['image'][0,0,:,:]
    label=  dat['label'][0,0,:,:]
    middleSlice = round(label.size()[2]/2)
    print(f"image shape: {image.shape}, label shape: {label.shape}")
    # plot the slice [:, :, 80]
    plt.figure("check", (12, 6))
    plt.subplot(1, 2, 1)
    plt.title("image")
    plt.imshow(image[:, :, middleSlice], cmap="gray")
    plt.subplot(1, 2, 2)
    plt.title("label")
    plt.imshow(label[:, :, middleSlice])
    plt.show()

```

```
Python 3.7
MONAI version: 0.9.dev2210
Numpy version: 1.21.5
Pytorch version: 1.10.2+cu113
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: 1a660e6a7a50e985af5ff76b559baab44175438c
MONAI __file__: d:\projects\vsCode\pythonForHtests\venv\lib\site-packages\monai\__init__.py

Optional dependencies:
Pytorch Ignite version: 0.4.8
Nibabel version: 3.2.2
scikit-image version: 0.19.2
Pillow version: 9.0.1
Tensorboard version: 2.8.0
gdown version: 4.4.0
TorchVision version: 0.11.3+cu113
tqdm version: 4.63.0
lmdb version: 1.3.0
psutil version: 5.9.0
pandas version: 1.1.5
einops version: 0.4.1
transformers version: 4.17.0
mlflow version: 1.24.0

```

full code 
https://github.com/jakubMitura14/pythonForHtests/blob/master/loadOnlyLiver.ipynb

[1] https://wiki.cancerimagingarchive.net/display/Public/CT-ORG%3A+CT+volumes+with+multiple+organ+segmentations

Thank you for Help !

**Comment by wyli** [comment, marked as answer]
thanks for the detailed report, I can replicate it with volume 20 and 21, the issue seems to be in the original metadata.


`volume-20.nii.gz` doesn't have the metadata about orientations (possibly not handled properly when converting from the source into a nifti file)
<details>

![Screenshot 2022-03-09 at 08 42 13](https://user-images.githubusercontent.com/831580/157404678-5a75f8e2-4f86-4fec-a71a-2130d360102a.png)

</details>

while `volume-21.nii.gz` looks fine:
<details>

![Screenshot 2022-03-09 at 08 41 02](https://user-images.githubusercontent.com/831580/157404500-b1194bb6-dee5-4116-a8b1-3b921f87ddb9.png)

</details>

**Reply by jakubMitura14** [reply]
Thank you for help !!! - Is there a way to filter out the images that do not hold proper spacing and orientation metadata ?

**Reply by wyli** [reply]
sure, you can use the `sform_code` and `qform_code` fields, if both of them are zeros, probably the metadata is not valid:
```py
>>> from monai.transforms import LoadImaged
>>> data_dict = LoadImaged(keys="img")({"img": "./volume-20.nii.gz"})
>>> (data_dict["img_meta_dict"]["qform_code"], data_dict["img_meta_dict"]["sform_code"])
(array(0, dtype=int16), array(0, dtype=int16))

>>> data_dict = LoadImaged(keys="img")({"img": "./volume-21.nii.gz"})
>>> (data_dict["img_meta_dict"]["qform_code"], data_dict["img_meta_dict"]["sform_code"])
(array(1, dtype=int16), array(1, dtype=int16))
```

**Reply by jakubMitura14** [reply]
Thank You so much !!
